runners/google-cloud-dataflow-java/worker/src/main/java/org/apache/beam/runners/dataflow/worker/fn/control/ElementCountMonitoringInfoToCounterUpdateTransformer.java

```diff
@@ -95,7 +99,12 @@ public CounterUpdate transform(MonitoringInfo monitoringInfo) {
       return null;
     }
 
-    long value = monitoringInfo.getMetric().getCounterData().getInt64Value();
+    long value;
+    try {
+      value = VARINT_CODER.decode(monitoringInfo.getPayload().newInput());
```

Comment:
Can we encapsulte this so that its still one line to get the value out? I think that this would be a nicer interface. For consumers. Ideally something as easy to use on the receiving side as SimpleMonitoringInfoBuilder.

i.e.
long value = MonitoringInfos.getSumInt64Value(monitoringInfo)

Reply:
Done here an elsewhere. I introduced a MonitoringInfoEncodings class with the convenience methods for the currently used encodings.